pkg/cli/clisqlclient/txn_shim.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package clisqlclient
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach-go/crdb"
+)
+
+// sqlTxnShim implements the crdb.Tx interface.
+//
+// It exists to support crdb.ExecuteInTxn. Normally, we'd hand crdb.ExecuteInTxn
+// a sql.Txn, but sqlConn predates go1.8's support for multiple result sets and
+// so deals directly with the lib/pq driver. See #14964.
+//
+// TODO(knz): This code is incorrect, see
+// https://github.com/cockroachdb/cockroach/issues/67261
+type sqlTxnShim struct {
+	conn *sqlConn
+}
+
+var _ crdb.Tx = sqlTxnShim{}
```

Comment:
i'm somewhat amused we import all of cockroach-go just to ensure it obeys the interface. maybe we should move this assertion to the test to optimise out this import.

Reply:
It's not just that, we actually use `crdb.ExecuteInTx` to implement `(*sqlConn).ExecTxn`.